yaksh/evaluator_tests/test_bash_evaluation.py

```diff
@@ -48,5 +51,81 @@ def test_infinite_loop(self):
         self.assertEquals(result.get("error"), self.timeout_msg)
 
 
+class BashStdioEvaluationTestCases(unittest.TestCase):
+    def setUp(self):
+        self.timeout_msg = ("Code took more than {0} seconds to run. "
+                            "You probably have an infinite loop in your"
+                            " code.").format(SERVER_TIMEOUT)
+
+    def test_correct_answer(self):
+        user_answer = dedent(""" #!/bin/bash
+                             read A
+                             read B
+                             echo -n `expr $A + $B`
+                             """
+                             )
+        test_case_data = [{'expected_output': '11', 'expected_input': '5\n6'}]
+        get_class = BashStdioEvaluator()
+        kwargs = {"user_answer": user_answer,
+                  "test_case_data": test_case_data
+                  }
+        result = get_class.evaluate(**kwargs)
+        self.assertEquals(result.get('error'), "Correct Answer")
+        self.assertTrue(result.get('success'))
+
+    def test_array_input(self):
+        user_answer = dedent(""" readarray arr;
+                                 COUNTER=0
+                                 while [  $COUNTER -lt 3 ]; do
+                                     echo -n "${arr[$COUNTER]}"
+                                     let COUNTER=COUNTER+1 
+                                 done
+                            """
+                             )
+        test_case_data = [{'expected_output': '1 2 3\n4 5 6\n7 8 9\n',
+                           'expected_input': '1,2,3\n4,5,6\n7,8,9'
+                           }]
+        get_class = BashStdioEvaluator()
+        kwargs = {"user_answer": user_answer,
+                  "test_case_data": test_case_data
+                  }
+        result = get_class.evaluate(**kwargs)
+        self.assertEquals(result.get('error'), "Correct Answer")
+        self.assertTrue(result.get('success'))
+
+    def test_incorrect_answer(self):
+        user_answer = dedent(""" #!/bin/bash
+                             read A
+                             read B
+                             echo -n `expr $A + $B`
+                             """
+                             )
+        test_case_data = [{'expected_output': '12', 'expected_input': '5\n6'}]
```

Comment:
This testcase is fine but would be better if you write a wrong code, like subtraction instead of addition.
The code is proper but expected output is something else, a bit not clear.
Expected output can be 11 and not 12.